src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -193,13 +196,26 @@ async def infer_python_conftest_dependencies(
     )
 
 
-def rules():
+# This is a separate function to facilitate tests registering import inference.
+def import_rules():
```

Comment:
I had a hell of a time getting all the tests to register the rule graph correctly :/

Reply:
Yea, that remains quite problematic... I think mostly because of the opacity of rule graph errors. Have opened #11254, because I don't think we're currently tracking it explicitly.